### PR TITLE
Split up System.Xml solution

### PR DESCRIPTION
**Pull request by Maxwe11**
As background for #746 `System.Xml.sln` should be splitted into:
- System.Xml.XDocument.sln
- System.Xml.XmlDocument.sln
- System.Xml.XPath.sln
- System.Xml.XPath.XDocument.sln
- System.Xml.XPath.XmlDocument.sln

### CODE REVIEW

File: src/System.Xml.XPath.XDocument/src/packages.config [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+  <package id="System.Linq" version="4.0.0-beta-22512" />
   <package id="System.Reflection" version="4.0.10-beta-22512" />
-  <package id="System.Threading" version="4.0.0-beta-22512" />
+  <package id="System.Resources.ResourceManager" version="4.0.0-beta-22512" />
+  <package id="System.Runtime" version="4.0.20-beta-22512" />
   <package id="System.Runtime.Extensions" version="4.0.10-beta-22512" />
-  <package id="System.Linq" version="4.0.0-beta-22512" />
+  <package id="System.Text.Encoding" version="4.0.10-beta-22512" targetFramework="portable-net45+win" />
+  <package id="System.Threading" version="4.0.0-beta-22512" />
+  <package id="System.Threading.Tasks" version="4.0.10-beta-22512" targetFramework="portable-net45+win" />
```
Comment (stephentoub) [inline src/System.Xml.XPath.XDocument/src/packages.config]: How come System.IO, System.Text.Encoding, and System.Threading.Tasks weren't necessary before but are now?

File: src/System.Xml.XPath.XmlDocument/src/packages.config [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+  <package id="System.Diagnostics.Tools" version="4.0.0-beta-22512" />
+  <package id="System.Globalization" version="4.0.10-beta-22512" />
+  <package id="System.IO" version="4.0.10-beta-22512" />
   <package id="System.Reflection" version="4.0.10-beta-22512" />
+  <package id="System.Resources.ResourceManager" version="4.0.0-beta-22512" />
+  <package id="System.Runtime" version="4.0.20-beta-22512" />
+  <package id="System.Runtime.Extensions" version="4.0.10-beta-22512" />
+  <package id="System.Text.Encoding" version="4.0.10-beta-22512" targetFramework="portable-net45+win" />
+  <package id="System.Threading" version="4.0.0-beta-22512" />
+  <package id="System.Threading.Tasks" version="4.0.10-beta-22512" targetFramework="portable-net45+win" />
```
Comment (stephentoub) [inline src/System.Xml.XPath.XmlDocument/src/packages.config]: Same question about System.Text.Encoding and System.Threading.Tasks...?

File: src/System.Xml.XPath.XDocument/src/packages.config (hunk 1 above)
Comment (Maxwe11) [inline src/System.Xml.XPath.XDocument/src/packages.config]: These dependencies were added by nuget.

1. `System.Xml.XPath.XDocument` depends on **`System.Xml.XDocument`** and `System.Xml.XPath`
2. **`System.Xml.XDocument`** depends on `System.IO` and **`System.Xml.ReaderWriter`**
3. **`System.Xml.ReaderWriter`** depends on `System.Text.Encoding` and `System.Threading.Tasks`

File: src/System.Xml.XPath.XmlDocument/src/packages.config (hunk 2 above)
Comment (Maxwe11) [inline src/System.Xml.XPath.XmlDocument/src/packages.config]: Same dependent on `System.Xml.ReaderWriter` requires `System.Text.Encoding` and `System.Threading.Tasks`

**Comment by krwq** [comment]
Thanks for this, looks good :+1: I saw some opinions that they prefer to have it together - I personally like this split and find it more convenient as XML project names are really similar.

**Comment by Maxwe11** [comment]
resolved

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**